aten/src/ATen/core/ivalue.cpp

```diff
@@ -61,10 +45,41 @@ std::ostream& operator<<(std::ostream & out, const List<IValue> & v) {
   return printList<IValue>(out, v, "(", ", ", ")");
 }
 
+
+//String, Bool, and Double are special-cased
+#define TORCH_FORALL_TAGS(_) \
+  _(None) \
+  _(Tensor) \
+  _(Int) \
+  _(Tuple) \
+  _(IntList) \
+  _(DoubleList) \
+  _(BoolList) \
+  _(TensorList) \
+  _(Blob) \
+  _(GenericList) \
+  _(World) \
+
 std::ostream& operator<<(std::ostream & out, const IValue & v) {
   switch(v.tag) {
     #define DEFINE_CASE(x) case IValue::Tag::x: return out << v.to ## x();
     TORCH_FORALL_TAGS(DEFINE_CASE)
+    case IValue::Tag::Bool: {
+      if (v.toBool())
+        out << "True";
+      else
+        out << "False";
+      return out;
+    }
+    case IValue::Tag::String: return out << "'" << v.toString() << "'";
+    case IValue::Tag::Double: {
+      auto dbl = v.toDouble();
+      out << dbl;
+      //distinguish doubles without fraction from ints
+      if (std::fmod(dbl, 1.0) == 0)
```

Comment:
This is not a good approach. If the number is big enough it will be formatted using scientific notation and this will format wrong.